dist/lucca-ui.js

```diff
@@ -872,6 +873,437 @@
+				showFormerEmployees: "=", // boolean
+				/*** HOMONYMS ***/
+				homonymsProperties: "@", // list of properties to handle homonyms
+				/*** CUSTOM FILTER ***/
+				customFilter: "&", // should be a function with this signature: function(user){ return boolean; } 
+				/*** OPERATION SCOPE ***/
+				appId: "@",
+				operation: "@"
+			},
+			link: function (scope, elt, attrs, ctrl) {
+				if (attrs.homonymsProperties) {
+					scope.properties = attrs.homonymsProperties.split(',');
+				}
+				else {
+					scope.properties = DEFAULT_HOMONYMS_PROPERTIES;
+				}
+				ctrl.isMultipleSelect = true;
+				ctrl.asyncPagination = false;
+				ctrl.useCustomFilter = !!attrs.customFilter;
+			}
+		};
+	})
+
+	.controller("luidUserPickerController", ['$scope', '$http', 'moment', '$timeout', '$q', function ($scope, $http, moment, $timeout, $q) {
+		var ctrl = this;
+		var promise; // store the current get request to fetch users
+		// Only used for UserPickerMultiple
+		var selectedUsersCount = 0;
+		// Only used for asynchronous pagination
+		var timeout = {}; // object that handles timeouts - timeout.count will store the id of the timeout related to the count query
+
+		//$scope.user = {};
+		$scope.selected = {};
+		$scope.selected.users = [];
+
+		/****************/
+		/***** FIND *****/
+		/****************/
+
+		$scope.find = function (clue) {
+			getUsersAsync(clue).then(
+				function(users) {
+					var filteredUsers = users;
+					var count = filteredUsers.length;
+
+					// Used for UserPickerMultiple only
+					if (ctrl.isMultipleSelect) {
+						// Remove duplicates between results and selected users
+						_.each($scope.selected.users, function (selectedUser) {
+							filteredUsers = _.reject(filteredUsers, function (user) {
+								return (user.id === selectedUser.id);
+							});
+						});
+						// Number of results matching the query
+						count = filteredUsers.length;
+					}
+
+					// Used when a custom filtering function is given
+					if (ctrl.useCustomFilter) {
+						// Get new set of users
+						filteredUsers = _.filter(filteredUsers, function(user){ return $scope.customFilter(angular.copy(user)); });
+						count = filteredUsers.length;
+					}
+
+					if (hasFormerEmployees(filteredUsers)) {
+						handleFormerEmployees(filteredUsers);
+					}
+
+					if (hasHomonyms(filteredUsers)) {
+						tagHomonyms(filteredUsers);
+						handleHomonyms(filteredUsers);
+					}
+
+					if (hasPagination(filteredUsers)) {
+						if (asyncPagination) {
+							filteredUsers = addOverflowMessage(filteredUsers, "...");
+							// Call the api to get the right count and update the message to the user
+							handlePaginationAsync(clue).then(function(cnt){
+								count = cnt;
+								filteredUsers = updateOverflowMessage(filteredUsers, MAX_COUNT, count);
+							}, function(error) {
+								// HANDLE ERROR
+							});
+						}
+						else {
+							filteredUsers = addOverflowMessage(filteredUsers, count);
+						}
```

Comment:
here just have this code block
```js
if(asyncPagination){
  handlePaginationAsync(clue)
  .catch(function(message){
    handleError(paginationError, message);
  });
}else{
  handlePagination();
}
```
what i mean is the method `find` is the conductor and it calls different methods, it should have very little code and all block should look like this
```js
if(hasSomeCriteria()){
  handleStuff();
}
```
or this if `handleStuff` is async
```js
if(hasSomeCriteria()){
  handleStuffAsync()
  .catch(function(message){
    // HANDLE ERROR
  }));
}
```

FYI the `.catch(callbackFun)` is a shortcut for `.then(function(){}, callbackFun)`. it's when you're only interested in the promise rejection